app/src/main/java/com/asfoundation/wallet/ui/iab/BillingWebViewFragment.kt

```diff
@@ -157,9 +162,28 @@ class BillingWebViewFragment : BasePageViewFragment() {
     view.webview.settings.domStorageEnabled = true
     view.webview.settings.useWideViewPort = true
     view.webview.loadUrl(currentUrl)
+
+    view?.warning_get_bt?.setOnClickListener {
+      dismissGetAppWarning()
+      currentExtAppSelected?.let {
+        openDownloadExternalApp(it)
+      }
+    }
+    view?.warning_cancel_bt?.setOnClickListener {
+      dismissGetAppWarning()
+    }
+
     return view
   }
 
+  private fun openDownloadExternalApp(appInfo: ExternalAppEnum) {
+    try {
+      startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(appInfo.marketUri)))
+    } catch (e: ActivityNotFoundException) {
+      startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(appInfo.googlePlayUrl)))
```

Comment:
This also can throw `ActivityNotFoundException` for Huawei devices for example, right?

Reply:
Added another try catch for that case. It could throw if the phone doesn't have any web browser installed.